tests/jobs/test_local_task_job.py

```diff
@@ -462,7 +463,7 @@ def test_failure_callback_called_by_airflow_run_raw_process(self, monkeypatch, t
         assert len(lines) == 1  # invoke once
         assert lines[0].startswith(ti.key.primary)
         this_pid = str(os.getpid())
-        assert this_pid not in lines[0]
+        assert not lines[0].endswith(f"pid: {this_pid}")
```

Comment:
This should probably be improved to test that the line _does_ end with something like `pid: {pid}`, but the `{pid}` part is not the current process.

Reply:
I thought it required some refactoring of test DAG https://github.com/apache/airflow/blob/main/tests/dags/test_on_failure_callback.py

Reply:
Nah I’m thinking just something like

```python
match = re.match(r"^.+pid: (\d+)")
assert match is not None
assert int(match.group(0)) != os.getpid()
```

Reply:
Brilliant. I was a little lazy last night.

Reply:
Actually I came up with similar fix as original simple one (in #24584) and somehow I have a a feeling this is a better one (well certainly simpler to understand). There is a this ancient Chinese proverb "You have a problem? Introduce regexp - you will have two problems"  :D. Somehow I feel checking endswith `: ID`.  is better (but I just realized that my solution was actually wrong as I did not check for EOL :).  

So I am ok with merging this one after rebase.